### DISCUSSION TITLE
Web Search tool for CLINE

### DISCUSSION THREAD

**Discussion by decentralize-all** [cline/cline#496] [Feature Requests]
As part of the agent tool kit it would be great have the ability for web search similar to Claude Engineer with Tavily or similar approach. Keeping current on the latest solutions is key for quality development and Claude 3.5 is nearly 2yrs old stale and its corpus is questionable when it comes to latest web3, genai, emerging tech code and patterns. Yes i know GPT has search as an alternative, that said i do think move of the agent behaviors in CLINE have been tuned for Claude to date & costs are dramatically different.

**Comment by decentralize-all** [comment]
@saoudrizwan any chance for this? :). Happy to see if i can figure it out if time permits. Would probably need some input though. Is it on your priority list for next 2m?

**Comment by elemental-mind** [comment]
I think this is solved with MCP now, right?

**Comment by saoudrizwan** [comment]
@elemental-mind is correct–this is now possible using MCP! https://x.com/sdrzn/status/1867271665086074969
